src/smt/theory_str_noodler/decision_procedure.h

```diff
@@ -339,7 +339,16 @@ namespace smt::noodler {
         /**
          * @brief Construct constraints to get rid of not_contains predicates.
          */
-        void replace_not_contains();
+        lbool replace_not_contains();
+
+        /**
+         * @brief Syntactically unify not contains terms. If they they are included (in the sense of vectors) the 
+         * not(contain) is unsatisfiable.
+         * 
+         * @param prep FormulaPreprocessor
+         * @return l_false -> unsatisfiable 
+         */
+        lbool can_unify_not_contains(const FormulaPreprocessor& prep);
```

Comment:
It shouldn't have a side effect. I updated the comment to be less misleading.

Reply:
Also, does it hold that if you can unify not contains terms, then it is unsatisfiable? Because then the function name is a bit weird, it should return true if we can unify them.

Reply:
You are right, it's a bit weird. I changed the semantics.